packages/graphile-build-pg/src/QueryBuilder.js

```diff
@@ -744,7 +753,14 @@ ${sql.join(
         : this.buildSelectFields();
 
     let fragment = sql.fragment`\
-select ${useAsterisk ? sql.fragment`${this.getTableAlias()}.*` : fields}
+select ${
+      this.compiledData.distinctOn.length &&
+      sql.fragment`distinct on (${sql.join(
+        this.compiledData.distinctOn,
+        ", "
+      )})`
```

Comment:
Please use ternary rather than passing `0` into the `sql` statement. We do happen to handle that currently because we have a truthy test, but that will go away in the next major version - it was an accident and it allows mistakes to pass through unchallenged.

```suggestion
      this.compiledData.distinctOn.length
      ? sql.fragment`distinct on (${sql.join(
          this.compiledData.distinctOn,
          ", "
        )})`
      : sql.blank
```